web/src/settings_invites.ts

```diff
@@ -197,13 +205,21 @@ function do_resend_invite({$row, invite_id}: {$row: JQuery; invite_id: string}):
         );
     }
 
-    $resend_button.prop("disabled", true).text($t({defaultMessage: "Working…"}));
```

Comment:
I guess we still want the disabled state of button as we do not allow resending the invite again. But I am not sure. @alya thoughts?

Currently in main, the button is disabled after re-sending the mail with button text being changed to "Sent!".

Reply:
I'm not sure I understand the full context, but the issue did not specify any changes to button states.

Reply:
Let me know if I'm missing something where such a change would make sense, though.

Reply:
The PR also changes the errors returned by server in the modal now which is consistent with the other places using modal. Previously the error was shown on the button, i.e. the resend/revoke button.

And previously the button text was changed to "Working..." and the button was disabled while the request was being processed. That is not needed now since modal is closed only after the request completes and till then loading spinner is shown on the modal button.

So, while making the change to no longer make the button text "Working...", the PR also changes the behavior of button once the "Resend" request is completed successfully. Previously the "Resend" button was disabled (background color of button is set to gray) with the text being changed to "Sent!" on successfully resending the invite. So, just wanted to confirm whether we want to keep that behavior or not.

Reply:
I have updated the PR, so that the button state(in both Resend and Revoke) remains disabled.
Following the above comment, I would like to state a few points.

When we resend an invitation, the following cases may occur:
- **Error due to non-matching fields:** This error is modified to display in the confirmation modal as stated in the issue. It is completed successfully.
- **Server error:** This error was shown in place of the Resend and Revoke buttons. As @sahil839 [mentioned ](https://github.com/zulip/zulip/pull/31896#pullrequestreview-2354556737) server errors should also be displayed in the confirmation modal, I have made changes so that server errors should also display in the confirmation modal. (Please see [this comment](https://github.com/zulip/zulip/pull/31896#issuecomment-2404899338) for changes in how server errors are handled as per this PR.)
- **Successful response from the server:**
  - In main: the Resend button is disabled after resending the invitation, with the button text being changed to "Sent!".
  - In this PR: I have removed the "Working..." text from the button, as the modal remains open while the request is processed, as per [this comment](https://github.com/zulip/zulip/pull/31896#discussion_r1798900111).

I believe we should maintain the disabled state of the Resend button. I have made this change in the PR and pushed it for testing and review.
Thank you.